src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -15,14 +15,15 @@ namespace Microsoft.Build.BuildCheck.Infrastructure;
 internal sealed class BuildCheckConnectorLogger : ILogger
 {
     private readonly Dictionary<Type, Action<BuildEventArgs>> _eventHandlers;
-    private readonly IBuildCheckManager _buildCheckManager;
+    private IBuildCheckManager _buildCheckManager;
+    private readonly IBuildCheckManager _buildCheckManagerForEnabledState;
```

Comment:
Btw. I definitely like this swapping approach.

I'm not 100% sure whether we want to have the infra dictate if restore is happening or have the other approach - storing the flag in BuildEvaluationStartedEventArg (as then the handling might be more unified for the replay case).
But even if we stay with BuildEvaluationStartedEventArg case - I'd like to adopt this swapping strategy